### PR TITLE
[KARAF-6321] - add configuration options to protect against CTRL-D/shell:logout exits

### PR DESCRIPTION
**Pull request by awrb**

### CODE REVIEW

File: shell/core/src/main/java/org/apache/karaf/shell/impl/console/ConsoleSessionImpl.java [hunk 1]
```diff
@@ -451,7 +449,12 @@ private CharSequence readCommand(AtomicBoolean reading) throws UserInterruptExce
                 command = reader.getBuffer().toString();
             }
         } catch (EndOfFileException e) {
-            command = null;
+            boolean disableEofExit = (boolean) session.get(Session.DISABLE_EOF_EXIT);
+            if (disableEofExit) {
+                command = "";
```
Comment (awrb) [inline shell/core/src/main/java/org/apache/karaf/shell/impl/console/ConsoleSessionImpl.java:454]: I tested it manually and it works fine but I couldn't figure out a way to write a test for this (e.g. how to send CTRL-D to Karaf in a pax exam test, or some other way)

**Review by jbonofre — COMMENTED** [review]
Overall, it looks good (I will take a deeper look). Do you mind to update the document (in the `manual` module) ?

**Comment by awrb** [comment]
Sure @jbonofre I will do it in a couple of hours. Actually some of my other pull requests might also need a change in the `manual`. 
Do you have any ideas how to code a test for the diasabled EOF? 


> Overall, it looks good (I will take a deeper look). Do you mind to update the document (in the `manual` module) ?

**Comment by awrb** [comment]
I updated `console.adoc` and `remote.adoc` in `manual`.

File: shell/core/src/main/java/org/apache/karaf/shell/impl/console/ConsoleSessionImpl.java (hunk 1 above)
Comment (jbonofre) [reply shell/core/src/main/java/org/apache/karaf/shell/impl/console/ConsoleSessionImpl.java:454]: That's actually pretty hard in pax-exam as it controls the lifecycle. Manual test is fine and enough IMHO.

**Review by jbonofre — APPROVED** [review]

**Review by jbonofre — APPROVED** [review]
Thanks for the documentation update! I will squash and merge.

**Pull request closed without merging**